llvm/include/llvm/MC/MCRegisterInfo.h

```diff
@@ -101,6 +102,9 @@ class MCRegisterClass {
 
   /// Return true if this register class has a defined BaseClassOrder.
   bool isBaseClass() const { return BaseClass; }
+  /// isSynthetic - Return true if this is a synthetic class. This field helps
+  /// targets to dynamically enable the regclass during codegen.
+  bool isSynthetic() const { return Synthetic; }
```

Comment:
What is the reasoning for the "syntethic" name? I feel like "dynamic" suits it more, no?
When I see synthetic I imagine a capability to generate a RC using C++, but these still need to be defined in .td

Reply:
We want to express a software constraint. Every other class is expressing a hardware constraint. We want to express a synthetic allocation window